webui/react/src/components/Logo.tsx

```diff
@@ -66,13 +69,18 @@ const Logo: React.FC<Props> = ({
     ].join();
   }, [branding]);
 
-  return (
-    <img
-      alt={alt}
-      className={classes.join(' ')}
-      src={logos[branding][orientation][isDarkMode ? 'dark' : 'light']}
-    />
-  );
+  useEffect(() => {
+    const mode = isDarkMode ? 'dark' : 'light';
+    if (hasCustomLogo) {
+      const imageUrl = `/det/customer-assets/logo?orientation=${orientation}&mode=${mode}`;
+      setImageSrc(imageUrl);
+      setClasses([css.customLogo]);
+    } else {
+      setImageSrc(logos[branding][orientation][mode]);
+    }
+  }, [branding, orientation, isDarkMode, hasCustomLogo]);
+
+  return <img alt={alt} className={classes.join(' ')} src={logoSrc} />;
```

Comment:
is there a reason to use `useEffect` and `useState` instead of just changing what we render based on the `hasCustomLogo` prop?

Reply:
no good reason. previously, it was doing some async work but not anymore. I updated it